src/tasks.coffee

```diff
@@ -117,9 +118,11 @@ processNextTaskRound = (task, callback) ->
 
       promises.push defer.promise
 
-  task.save ->
```

Comment:
I needed to move this here for some of the tests to pass, otherwise the task wasn't being saved at the correct time. I guess it should be in both places then, I'l try that.